projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/cached_data_job_executor.py

```diff
@@ -127,3 +133,78 @@ def get_all_jobs(self):
+        """
+        Start a new data job execution.
+        The stages of the process are:
+            1) Get the latest available execution_id before any new
+               executions have been started.
+            2) Start a new execution.
+            3) In case of a Timeout exception, check if a new execution
+               has been started and re-try step 2 if needed.
+
+        NOTE: A loop is used to handle situations, where due to some
+        network instability, a socket timeout happens. The execution of
+        the data job may have started, but we don't know because an execution
+        id was not returned due to the exception.
+
+        :param job_name: name of the data job to be executed
+        :param team_name: name of the owning team
+        :return: id of the started job execution
+        """
+        current_retries = 0
+        execution_id = None
+
+        latest_available_execution_id = self.get_latest_available_execution_id(
+            job_name=job_name, team=team_name
+        )
+
+        while current_retries < ALLOWED_RETRIES:
+            try:
+                execution_id = self._executor.start_job(job_name, team_name)
+                return execution_id
+            except TimeoutError as e:
+                log.info(
+                    f"A timeout exception occurred while starting the {job_name} data job. "
+                    f"Exception was: {e}"
+                )
+
+            if not execution_id:
+                execution_id = self.get_latest_available_execution_id(
+                    job_name=job_name, team=team_name
+                )
+            if execution_id and execution_id != latest_available_execution_id:
+                return execution_id
+            else:
+                current_retries += 1
+                time.sleep(SLEEP_TIME)  # Sleep for 10 seconds before re-try.
+
+        # If the execution reaches this point, then something has happened,
+        # and the state of the data job that has been started cannot be determined.
+        raise RuntimeError(
+            "Bug, fix me! Something wrong has happened and I cannot recover!"
+        )
+
+    def get_latest_available_execution_id(
+        self, job_name: str, team: str
+    ) -> Optional[str]:
+        """
+        Get the latest execution_id from the list of all executions for a
+        given data job.
+
+        :param job_name: name of data job for which an execution_id is needed
+        :param team: name of the team owning the data job
+        :return: The execution_id of the latest job execution or None if no
+                 executions are available.
+        """
+        latest_execution_id = None
+        executions_list = self._executor.job_executions_list(
+            job_name=job_name, team_name=team
+        )
+
+        if executions_list:
+            # We need only the latest execution which is the last element of the list
+            latest_execution = executions_list[-1]
```

Comment:
Do you really need to create the `latest_execution` variable? can you not just do `executions_list[-1].id`

Reply:
Oopsie, good catch! It was not necessary. Fixed, thanks